### ISSUE TITLE
'connect ECONNREFUSED 127.0.0.1:4444' with WebdriverIO v5

### ISSUE THREAD

**Issue by prathameshnyt** [stsvilik/wdio-docker-service#53]
We are getting following error with v5 version of WebdriverIO.

```
[0-0] RUNNING in chrome - /test/basicTest.js
[0-0] This test will run using docker service provided by WebDriverIO using standalone-chrome-debug image.
[0-0] 2019-06-07T18:34:22.521Z ERROR @wdio/runner: Error: connect ECONNREFUSED 127.0.0.1:4444
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1097:14)
0-0 worker error { name: 'Error',
  message: 'connect ECONNREFUSED 127.0.0.1:4444',
  stack:
   'Error: connect ECONNREFUSED 127.0.0.1:4444\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1097:14)' }
[0-0] FAILED in chrome - /test/basicTest.js

Spec Files:	 0 passed, 1 failed, 1 total (100% completed) in 00:00:00
``` 

Node Version: v10.15.3
@wdio/local-runner@5.9.6
wdio-docker-service@2.1.0

```
services = ['docker'];
    host = 'localhost';
    dockerLogs = './dockerLogs';
    dockerOptions = {
        image: 'selenium/standalone-chrome-debug',
        healthCheck: {
            url: 'https://localhost:4444',
            maxRetries: 3,
            inspectInterval: 1000,
            startDelay: 15000
        },
        options: {
            p: ['4444:4444'],
            shmSize: '2g'
        }
    };
```

Can you please let us know if we are missing something?

**Comment by stsvilik** [comment]
@prathameshnyt could you please provide docker version and a full config file. This error indicates that selenium docker container has not started at the time of the test. This could also be due to test code as well. Do you mind sharing `/test/basicTest.js` file?

**Comment by prathameshnyt** [comment]
@stsvilik thanks for looking into it. Yes, selenium docker container is not starting for some reason. If I run `docker run hello-world` it works fine.

```
docker version
Client: Docker Engine - Community
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:39 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:13:06 2019
  OS/Arch:          linux/amd64
  Experimental:     true
```

`test/basicTest.js` file:
```
describe('webdriver.io page', () => {
    it('should have the right title', () => {
        browser.url('https://webdriver.io');
        const title = browser.getTitle();
        assert.equal(title, 'WebdriverIO · Next-gen WebDriver test framework for Node.js');
    });
});
```

**Comment by prathameshnyt** [comment]
Mac OS Version: 10.14.5 (18F132)

Can you please share your working WDIO, docker version, and config file? I will compare it with mine.

Thanks,
Prathamesh

**Comment by stsvilik** [comment]
You can find it in the source where integration tests are. I believe that your code is failing because you don't `await` for browser to open a url.

New webdriverio does not include `sync` by default, so you need to use `await/async` to make things work.

Change your code to this and try:

```js
describe('webdriver.io page', () => {
    it('should have the right title', async () => {
        await browser.url('https://webdriver.io');
        const title = await browser.getTitle();
        assert.equal(title, 'WebdriverIO · Next-gen WebDriver test framework for Node.js');
    });
});
```

**Comment by prathameshnyt** [comment]
I tried this. I am seeing the same error.

**Comment by prathameshnyt** [comment]
Thanks, I tried your wdio.conf from integration test and it worked for me without adding `await/async`. I will look into the diff and will update you. 

Thanks,
Prathamesh

**Comment by prathameshnyt** [comment]
Thanks, our bad. I was able to get it to work. We have if/else conditions in our wdio.conf to check run environments. We checked for e.g. environment is docker vs standalone chrome we define respective services and options. In this case, we were setting `dockerOptions` in if condition for docker but we have not added it in `exports.config`. Adding `dockerOptions: dockerOptions` worked. Thank you so much for your help in looking into it.
